### PR TITLE
closes #105: delete collection command implementation

### LINKED ISSUE #105: JSON API deleteCollection operation
add operation to delete a collection to the JSON API. This will facilitate automated testing such as nosqlbench tests

### PR DESCRIPTION
**Pull request by ivansenic**
**What this PR does**:
Implements delete collection command.

**Which issue(s) this PR fixes**:
Fixes #105

**Checklist**
- [x] Tests
- [x] Swagger updates

### CODE REVIEW

**Comment by maheshrajamani** [comment]
LGTM. Not part of this PR,  but need to document what is acceptable for namespace and collection name? Can we add it as validation in the command deserialization and add to its description?

**Review by maheshrajamani — APPROVED** [review]

**Comment by ivansenic** [comment]
> LGTM. Not part of this PR, but need to document what is acceptable for namespace and collection name? Can we add it as validation in the command deserialization and add to its description?

Yes, good point, we can add as part of the validation. Created a ticket: https://github.com/stargate/jsonapi/issues/149

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/DeleteCollectionCommand.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * @param name Name of the collection
+ */
+@Schema(description = "Command that deletes a collection if one exists.")
+@JsonTypeName("deleteCollection")
+public record DeleteCollectionCommand(
+    @NotBlank
+        @Schema(
+            description = "Name of the collection",
+            implementation = Object.class,
+            type = SchemaType.OBJECT)
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/DeleteCollectionCommand.java:21]: Shouldn't this be `String`?

Comment (ivansenic) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/DeleteCollectionCommand.java:21]: Yup fixing this in a followup PR. I was copying from the create collection command, which also has it wrong.

Comment (ivansenic) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/DeleteCollectionCommand.java:21]: https://github.com/stargate/jsonapi/pull/157

**Pull request merged**